pkg/sypgp/sypgp_test.go

```diff
@@ -197,6 +199,152 @@ func TestPushPubkey(t *testing.T) {
 	}
 }
 
+func TestEnsureDirPrivate(t *testing.T) {
+	tmpdir, err := ioutil.TempDir("", "test-ensure-dir-private")
+	if err != nil {
+		t.Fatalf("Cannot create temporary directory")
+	}
+	defer os.RemoveAll(tmpdir)
+
+	cases := []struct {
+		name        string
+		preFunc     func(string) error
+		entry       string
+		expectError bool
+	}{
+		{
+			name:        "non-existent directory",
+			entry:       "d1",
+			expectError: false,
+		},
+		{
+			name:        "directory exists",
+			entry:       "d2",
+			expectError: false,
+			preFunc: func(d string) error {
+				if err := os.MkdirAll(d, 0777); err != nil {
+					return err
+				}
+				return os.Chmod(d, 0777)
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		testEntry := filepath.Join(tmpdir, tc.entry)
+
+		if tc.preFunc != nil {
+			if err := tc.preFunc(testEntry); err != nil {
+				t.Errorf("Unexpected failure when calling prep function: %+v", err)
+				continue
+			}
+		}
+
+		err := ensureDirPrivate(testEntry)
+		switch {
+		case !tc.expectError && err != nil:
+			t.Errorf("Unexpected failure when calling ensureDirPrivate(%q): %+v", testEntry, err)
+
+		case tc.expectError && err == nil:
+			t.Errorf("Expecting failure when calling ensureDirPrivate(%q), but got none", testEntry)
+
+		case tc.expectError && err != nil:
+			t.Logf("Expecting failure when calling ensureDirPrivate(%q), got: %+v", testEntry, err)
+			continue
+
+		case !tc.expectError && err == nil:
+			// everything ok
+
+			fi, err := os.Stat(testEntry)
```

Comment:
You might be able to use `IsDir()` from `internal/pkg/util/fs/helper.go` to keep the code short and neat.

Reply:
since is testing for the file mode (see a couple of lines after this) I still have to call os.Stat and I would be only changing a call to `fileinfo.IsDir` to a call to `fs.IsDir`, but I'll keep in mind this package in the future for other cases, thanks.